src/ds/prob/bloom-cuckoo.md

```diff
@@ -0,0 +1,101 @@
+
+First, create a bloom filter and configure an acceptable false positive rate for your use case.
+You can also specify an initial capacity; the size of the dataset that you expect to add to the filter. Bloom filters can be configured to expand when this capacity is reached - [see the `BF.RESERVE` documentation for details](https://redis.io/commands/bf.reserve/).
+
+```redis Create a Bloom filter
+BF.RESERVE user:778:bought_products 0.001 50 // create new bloom filter at key "user:778:bought_products" with a desired false positive rate of 0.1% (0.001) and anticipated data set size of 50 entries
+```
+
+Next, add some products for user 778.
+
+```redis Add all bought product IDs to a Bloom filter
+BF.MADD user:778:bought_products 4545667 9026875 3178945 4848754 1242449 // Add five items to user 778's list
+```
+
+Next, run a couple of queries.
+
+```redis Has a user bought this product?
+BF.EXISTS  user:778:bought_products 1234567  // No, the user has not bought this product
+BF.EXISTS  user:778:bought_products 3178945  // The user might have bought this product
+BF.MEXISTS user:778:bought_products 01234567 4545667 // determines if one or more items exist in the filter
+```
+
+You can get information about your bloom filter using the `BF.INFO` command:
+
+```redis Information about the filter
+BF.INFO user:778:bought_products // returns information about the bloom filter at key "user:778:bought_products"
+```
+
+Inserting items into a Bloom filter that doesn't yet exist will create the filter for you.
+
+```redis Create and add items simultaneously
+BF.INSERT bloomFilter ITEMS foo bar baz // creates the "bloomFilter" key and adds three items to it, if the filter does not already exist
+BF.INSERT newBloomFilter CAPACITY 10000 ITEMS hello waves goodbye // creates the "newBloomFilter" key and adds three items to it, if the filter does not already exist
+BF.INSERT bloomF NOCREATE ITEMS foo bar // tries to add 2 items to a filter with an error if the filter does not already exist
+```
+
+You can read more about Bloom filters and their use cases [here](https://redis.io/docs/data-types/probabilistic/bloom-filter/). See [here](https://redis.io/commands/?group=bf) for the complete list of Bloom filter commands.
+
+## Cuckoo Filters
+
+[Cuckoo filters](https://en.wikipedia.org/wiki/Cuckoo_filter), which are similar to Bloom filters, are used to determine, with a high degree of certainty, whether or not an element is a member of a set. Unlike Bloom filters, you can delete items from a cuckoo filter.
```

Comment:
Cuckoo filters, though to a limited degree, allow checking if an element appears more than once (see `CF.COUNT`), hence, the term `set` is not correct here.